src/rtl/hbregex.c

```diff
@@ -260,7 +260,8 @@ static HB_BOOL hb_regex( int iRequest )
    }
    pRegEx = hb_regexGet( hb_param( 1, HB_IT_ANY ),
                          ( ! hb_parldef( 3, 1 ) ? HBREG_ICASE : 0 ) |
-                         ( hb_parl( 4 ) ? HBREG_NEWLINE : 0 ) );
+                         ( hb_parl( 4 ) ? HBREG_NEWLINE : 0 ) |
+                         ( hb_parl( 8 ) ? HBREG_DOTALL : 0 ));
```

Comment:
A space is needed between the parenthesis.

Why do you need a dedicated and hard-wired Harbour option for that, 
when you can enable `PCRE_DOTALL` option by adding `(?s)` to the search expression?